apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/PartDEventTransformerTest.java

```diff
@@ -30,6 +30,58 @@
    */
   @Test
   public void transformSampleARecord() throws FHIRException {
+    PartDEvent claim = getPartDEventClaim();
+    ExplanationOfBenefit eob = PartDEventTransformer.transform(new MetricRegistry(), claim);
+    assertMatches(claim, eob);
+  }
+
+  @Test
+  public void transformSampleARecordWithNPI() throws FHIRException {
+    String serviceProviderIdQualiferCode = "01";
+    String serviceProviderCode = TransformerConstants.CODING_NPI_US;
+    checkOrgAndFacility(serviceProviderIdQualiferCode, serviceProviderCode);
+  }
+
+  @Test
+  public void transformSampleARecordWithUPIN() throws FHIRException {
+    String serviceProviderIdQualiferCode = "06";
+    String serviceProviderCode = TransformerConstants.CODING_UPIN;
+    checkOrgAndFacility(serviceProviderIdQualiferCode, serviceProviderCode);
+  }
+
+  @Test
+  public void transformSampleARecordWithNCPDP() throws FHIRException {
+    String serviceProviderIdQualiferCode = "07";
+    String serviceProviderCode = TransformerConstants.CODING_NCPDP;
+    checkOrgAndFacility(serviceProviderIdQualiferCode, serviceProviderCode);
+  }
+
+  @Test
+  public void transformSampleARecordWithStateLicenseNumber() throws FHIRException {
+    String serviceProviderIdQualiferCode = "08";
+    String serviceProviderCode = TransformerConstants.CODING_STATE_LICENSE;
+    checkOrgAndFacility(serviceProviderIdQualiferCode, serviceProviderCode);
+  }
+
+  @Test
+  public void transformSampleARecordWithFederalTaxNumber() throws FHIRException {
+    String serviceProviderIdQualiferCode = "11";
+    String serviceProviderCode = TransformerConstants.CODING_FEDERAL_TAX_NUM;
+    checkOrgAndFacility(serviceProviderIdQualiferCode, serviceProviderCode);
+  }
+
+  private void checkOrgAndFacility(
+      String ServiceProviderIdQualiferCode, String ServiceProviderCode) {
```

Comment:
I'd suggest adding JavaDoc for this method to make it clearer what it does. Also: you should probably fix the case on its params, so that they don't look like types.